src/lib/zkapp.ts

```diff
@@ -203,7 +237,10 @@ function wrapMethod(
             let result: unknown;
             try {
               let clonedArgs = actualArgs.map(cloneCircuitValue);
-              result = method.apply(this, clonedArgs);
+              result = await assertPromise(
+                method.apply(this, clonedArgs),
+                noPromiseError
+              );
```

Comment:
throughout the wrapped method, we are now awaiting the method result and also throwing a helpful runtime error if it's not returning a promise